pkg/controller/rollout/batchrelease/inner_batchrelease.go

```diff
@@ -188,113 +183,22 @@ func (r *innerBatchRelease) Promote(index int32, isRollback, checkReady bool) (b
-				if err = r.Get(context.TODO(), types.NamespacedName{Namespace: r.rollout.Namespace, Name: dName}, obj); err != nil {
-					return err
-				}
-				obj.Spec.Paused = false
-				return r.Update(context.TODO(), obj)
-			})
-			if err != nil {
-				klog.Errorf("update rollout(%s/%s) stable deployment failed: %s", r.rollout.Namespace, r.rollout.Name, err.Error())
-				return false, err
-			}
-			klog.Infof("resume rollout(%s/%s) stable deployment(paused=false) success", r.rollout.Namespace, r.rollout.Name)
-		}
-
-		// Whether to wait for pods are ready
-		if !checkReady {
-			return true, nil
-		}
-		data := util.DumpJSON(obj.Status)
-		// wait for all pods are ready
-		maxUnavailable, _ := intstr.GetScaledValueFromIntOrPercent(obj.Spec.Strategy.RollingUpdate.MaxUnavailable, int(*obj.Spec.Replicas), true)
-		if obj.Status.ObservedGeneration != obj.Generation || obj.Status.UpdatedReplicas != *obj.Spec.Replicas ||
-			obj.Status.Replicas != *obj.Spec.Replicas || *obj.Spec.Replicas-obj.Status.AvailableReplicas > int32(maxUnavailable) {
-			klog.Infof("rollout(%s/%s) stable deployment status(%s), and wait a moment", r.rollout.Namespace, r.rollout.Name, data)
-			return false, nil
-		}
-		klog.Infof("resume rollout(%s/%s) stable deployment(paused=false) status(%s) success", r.rollout.Namespace, r.rollout.Name, data)
-		return true, nil
-
-	default:
-		// statefulset-like workloads
-		workloadRef := r.rollout.Spec.ObjectRef.WorkloadRef
-		workloadNsn := types.NamespacedName{Namespace: r.rollout.Namespace, Name: workloadRef.Name}
-		workloadGVK := schema.FromAPIVersionAndKind(workloadRef.APIVersion, workloadRef.Kind)
-		obj := &unstructured.Unstructured{}
-		obj.SetGroupVersionKind(workloadGVK)
-		err := r.Get(context.TODO(), workloadNsn, obj)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				klog.Warningf("rollout(%s/%s) statefulset(%s) not found, and return true", r.rollout.Namespace, r.rollout.Name, workloadNsn.Name)
-				return true, nil
-			}
-			return false, err
-		}
-
-		if util.GetStatefulSetPartition(obj) == 0 {
+func (r *innerBatchRelease) cancelBatchRelease(checkReady bool) (bool, error) {
+	batch, err := r.FetchBatchRelease()
+	if err != nil {
+		if errors.IsNotFound(err) {
 			return true, nil
 		}
+		return false, err
+	}
 
-		cloneObj := obj.DeepCopy()
-		body := fmt.Sprintf(`{"spec":{"updateStrategy":{"rollingUpdate":{"partition":0}}}}`)
-		err = r.Patch(context.TODO(), cloneObj, client.RawPatch(types.MergePatchType, []byte(body)))
-		if err != nil {
-			klog.Errorf("patch rollout(%s/%s) statefulset failed: %s", r.rollout.Namespace, r.rollout.Name, err.Error())
-			return false, err
-		}
-		klog.Infof("resume rollout(%s/%s) statefulset(partition=0) success", r.rollout.Namespace, r.rollout.Name)
-		return true, nil
+	if !batch.Spec.Strategy.Terminated {
+		body := fmt.Sprintf(
+			`{"spec":{"strategy":{"terminated":true,"terminationPolicy":{"type":"Promote","waitPromoted":%v}}}}`, checkReady)
```

Comment:
can we just set the batchrelease.batchPartition to nil, the terminationPolicy is quite confusing